lib/api/src/engine.rs

```diff
@@ -0,0 +1,92 @@
+use core::ops::Deref;
+
+#[cfg(feature = "sys")]
+use crate::sys::engine as engine_imp;
+#[cfg(feature = "sys")]
+pub(crate) use crate::sys::engine::default_engine;
+
+#[cfg(feature = "js")]
+use crate::js::engine as engine_imp;
+#[cfg(feature = "js")]
+pub(crate) use crate::js::engine::default_engine;
+
+/// The engine type
+#[derive(Clone, Debug)]
+pub struct Engine(pub(crate) engine_imp::Engine);
+
+impl Engine {
+    #[deprecated(
+        since = "4.0.0",
```

Comment:
Mmmm, so you want the 3.2 to become 4.0 it seems?

Reply:
Good catch. Let's make it 3.2

Reply:
```suggestion
        since = "3.2.0",
```